conbench/entities/commit.py

```diff
@@ -147,6 +152,9 @@ def commit_ancestry_query(self) -> Query:
 
     @staticmethod
     def create_no_context():
+        # Special commit row, a singleton that call results can relate to (in
+        # DB, via forgeign key) that have _no_ commit information set. But: is
+        # that needed? Why have that relation at all then?
```

Comment:
This is probably for the case where someone wants to use Conbench without supplying any `git` information. That use case is very limited in features but theoretically still possible.

Reply:
Right, question is if we need a special commit row to represent this case (and that the result can be db-linked with, via foreign key), or if that foreign key relationship is allowed to not exist. Not sure if I use the right terminology here.

Reply:
Oh I think I see what you're saying. Right now, we have this:
https://github.com/conbench/conbench/blob/f646b2b0a4c5ab3364bf41a8a35a8fe541e97b38/conbench/entities/run.py#L62
The question is, can we make this foreign key nullable, and use a NULL value when commit information doesn't exist, instead of creating a "dummy" commit with no information? Yes, I think that's probably better practice.

Reply:
Thanks for confirming!

Reply:
changing this here: https://github.com/conbench/conbench/pull/1134